model/task_lifecycle.go

```diff
@@ -470,62 +470,76 @@ func DeactivatePreviousTasks(ctx context.Context, t *task.Task, caller string) e
 	return nil
 }
 
+type stepbackInstructions struct {
+	shouldStepback bool
+	// If true, bisect should be used, if false, linear should be used.
+	bisect bool
+}
+
 // Returns true if the task should stepback upon failure, and false
 // otherwise. Note that the setting is obtained from the top-level
 // project, if not explicitly set on the task or disabled at the project level.
-func getStepback(taskId string) (bool, error) {
+func getStepback(taskId string) (stepbackInstructions, error) {
 	t, err := task.FindOneId(taskId)
 	if err != nil {
-		return false, errors.Wrapf(err, "finding task '%s'", taskId)
+		return stepbackInstructions{}, errors.Wrapf(err, "finding task '%s'", taskId)
 	}
 	if t == nil {
-		return false, errors.Errorf("task '%s' not found", taskId)
+		return stepbackInstructions{}, errors.Errorf("task '%s' not found", taskId)
 	}
 	projectRef, err := FindMergedProjectRef(t.Project, "", false)
 	if err != nil {
-		return false, errors.Wrapf(err, "finding merged project ref for task '%s'", taskId)
+		return stepbackInstructions{}, errors.Wrapf(err, "finding merged project ref for task '%s'", taskId)
 	}
 	if projectRef == nil {
-		return false, errors.Errorf("project for task '%s' not found", taskId)
+		return stepbackInstructions{}, errors.Errorf("project ref for task '%s' not found", taskId)
 	}
 	// Disabling the feature at the project level takes precedent.
 	if projectRef.IsStepbackDisabled() {
-		return false, nil
+		return stepbackInstructions{}, nil
 	}
 
 	project, err := FindProjectFromVersionID(t.Version)
 	if err != nil {
-		return false, errors.WithStack(err)
+		return stepbackInstructions{}, errors.WithStack(err)
 	}
 
 	projectTask := project.FindProjectTask(t.DisplayName)
 	// Check if the task overrides the stepback policy specified by the project
+	s := stepbackInstructions{}
+	if projectRef.StepbackBisect != nil {
```

Comment:
Can just do `s.bisect = utility.FromBoolPtr(projectRef.StepbackBisect)` and forgo the if statement here